src/re/AvatarGenerator.re

```diff
@@ -98,45 +73,37 @@ let make = (~randomize, ~settings, ~onChange, ~onExport) => {
     <div className="AvatarGenerator-row">
       {Belt.Array.map(settings, o =>
          <Styler
-           key={o.id}
+           key={o.label}
            id={o.id}
            label={o.label}
            colors={o.colors}
            styles={o.styles}
            selectedColor={o.selectedColor}
            selectedStyle={o.selectedStyle}
            onSelectColor={color => {
-             let key =
-               switch (componentFromJs(o.id)) {
-               | None => ""
-               | Some(component) =>
-                 switch (component) {
-                 | `Skin => "skinColor"
-                 | `Hair => "hairColor"
-                 | `FacialHair => "facialHairColor"
-                 | `Body => "bodyColor"
-                 | `Background => "bgColor"
-                 | _ => ""
-                 }
+             let key: Types.key =
+               switch (o.id) {
+               | `Skin => `SkinColor
+               | `Hair => `HairColor
+               | `FacialHair => `FacialHairColor
+               | `Body => `BodyColor
+               | `Background => `BgColor
+               | _ => Js.Exn.raiseError("ColorNotFound")
```

Comment:
TODO add [@bs.as] annotations for `id` polymorphic variant so we know which id the colors/styles weren't found for